src/main/java/org/terasology/gookeeper/system/BreedingSystem.java

```diff
@@ -165,11 +172,82 @@ public void onBreedingGooey(SelectForBreedingEvent event, EntityRef gooeyEntity,
      */
     @ReceiveEvent
     public void onBeginBreedingProcess(BeginBreedingEvent event, EntityRef gooeyEntity, GooeyComponent gooeyComponent) {
-        Prefab hatchlingPrefab = prefabManager.getPrefab("GooKeeper:gooeyHatchling");
-        EntityBuilder entityBuilder = entityManager.newBuilder(hatchlingPrefab);
-        LocationComponent locationComponent = entityBuilder.getComponent(LocationComponent.class);
-        LocationComponent gooeyLocation = gooeyEntity.getComponent(LocationComponent.class);
-        locationComponent.setWorldPosition(gooeyLocation.getWorldPosition().add(new Vector3f(2f, 1f, 2f)));
-        entityBuilder.build();
+        delayManager.addDelayedAction(gooeyEntity, Constants.spawnGooeyEggEventID, random.nextLong(4000, 7000));
+        AggressiveComponent aggressiveComponent = new AggressiveComponent();
+        FriendlyComponent friendlyComponent = new FriendlyComponent();
+        NeutralComponent neutralComponent = new NeutralComponent();
+
+        MatingComponent matingComponent = gooeyEntity.getComponent(MatingComponent.class);
+
+        AggressiveComponent parent1Aggressive = gooeyEntity.getComponent(AggressiveComponent.class);
+        AggressiveComponent parent2Aggressive = matingComponent.matingWithEntity.getComponent(AggressiveComponent.class);
+
+        if (parent1Aggressive != null) {
+            aggressiveComponent.aggressivenessFactor += parent1Aggressive.aggressivenessFactor;
+        }
+
+        if (parent2Aggressive != null) {
+            aggressiveComponent.aggressivenessFactor += parent2Aggressive.aggressivenessFactor;
+        }
+
+        aggressiveComponent.aggressivenessFactor /= 2f;
+
+        NeutralComponent parent1Neutral = gooeyEntity.getComponent(NeutralComponent.class);
+        NeutralComponent parent2Neutral = matingComponent.matingWithEntity.getComponent(NeutralComponent.class);
+
+        if (parent1Neutral != null) {
+            neutralComponent.neutralityFactor += parent1Neutral.neutralityFactor;
+        }
+
+        if (parent2Neutral != null) {
+            neutralComponent.neutralityFactor += parent2Neutral.neutralityFactor;
+        }
+
+        neutralComponent.neutralityFactor /= 2f;
+
+        FriendlyComponent parent1Friendly = gooeyEntity.getComponent(FriendlyComponent.class);
+        FriendlyComponent parent2Friendly = matingComponent.matingWithEntity.getComponent(FriendlyComponent.class);
+
+        if (parent1Friendly != null) {
+            friendlyComponent.friendlinessFactor += parent1Friendly.friendlinessFactor;
+        }
+
+        if (parent2Friendly != null) {
+            friendlyComponent.friendlinessFactor += parent2Friendly.friendlinessFactor;
+        }
+
+        friendlyComponent.friendlinessFactor /= 2f;
+
+        Component dominantComponent = getDominantComponent(aggressiveComponent, neutralComponent, friendlyComponent);
```

Comment:
Additionally to my above comment, you may find it easier to have a single component with a scale from aggression to friendliness.
Something like `AttitudeComponent` with a single float/int value. Use negatives as aggression and positives as friendliness, and zero as neutral (or rather, being within a set distance of zero)
Then you can just add the two parent's values, and you'll get the kids value.